Sling 6740 HTL Blog sample

auniverseaway opened the pull request:
This is the HTL blog sample I mentioned in [SLING-6740](https://issues.apache.org/jira/browse/SLING-6740). It's basic by design. In some scenarios it's less DRY to allow for easier understanding for a beginner. The README will have all the details to get a beginner started.

If there's any formatting or syntax constraints I missed, please let me know. If there's anything I should change to match the Sling project, let me know. Of course, any general tips are also appreciated.

Thanks!
Chris

![screen-shot](https://cloud.githubusercontent.com/assets/1972095/24581915/030c6694-16e2-11e7-9950-507d6cb0e2ec.png)

auniverseaway committed:
Initial commit of HTL Blog Sample

auniverseaway committed:
Working blog. Needs a bit of polish.

auniverseaway committed:
Adding created time. More polish and clean up.

auniverseaway committed:
More polish. Added readme.

auniverseaway committed:
Update readme.md

auniverseaway committed:
Adding comments to Edit model

auniverseaway committed:
More comments.

auniverseaway committed:
More formatting cleanup in prep for PR

auniverseaway committed:
Comments and a screenshot

auniverseaway committed:
Adding screenshot to Readme.

royteeuwen commented on samples/htlblog/pom.xml:
```diff
@@ -0,0 +1,201 @@
+        <plugins>
+            <plugin>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-scr-plugin</artifactId>
```
Remove this deprecated plugin and update the bundle plugin to parse the osgi ds annotations

royteeuwen commented on samples/htlblog/pom.xml:
```diff
@@ -0,0 +1,201 @@
+    <parent>
+        <groupId>org.apache.sling</groupId>
+        <artifactId>sling</artifactId>
+        <version>22</version>
```
you could use the latest version of the parent, it's already at 30

royteeuwen commented on samples/htlblog/pom.xml:
```diff
@@ -0,0 +1,201 @@
+    <dependencies>
+        <dependency>
+            <groupId>javax.servlet</groupId>
+            <artifactId>servlet-api</artifactId>
```
update servlet api to 3.1

royteeuwen commented on samples/htlblog/pom.xml:
```diff
@@ -0,0 +1,201 @@
+        </dependency>
+        <dependency>
+            <groupId>org.osgi</groupId>
+            <artifactId>org.osgi.compendium</artifactId>
```
old versions of the osgi libraries

royteeuwen commented on samples/htlblog/pom.xml:
```diff
@@ -0,0 +1,201 @@
+        <dependency>
+            <groupId>org.apache.sling</groupId>
+            <artifactId>org.apache.sling.models.api</artifactId>
+            <version>1.1.0</version>
```
you could use 1.3 here

royteeuwen commented on samples/htlblog/pom.xml:
```diff
@@ -0,0 +1,201 @@
+        </dependency>
+        <dependency>
+            <groupId>org.apache.jackrabbit</groupId>
+            <artifactId>jackrabbit-jcr-commons</artifactId>
```
why is this dep needed? you arent using it

royteeuwen commented on samples/htlblog/pom.xml:
```diff
@@ -0,0 +1,201 @@
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.jackrabbit</groupId>
```
why is this dep needed? you arent using it

royteeuwen commented on samples/htlblog/pom.xml:
```diff
@@ -0,0 +1,201 @@
+        </dependency>
+        <dependency>
+            <groupId>org.apache.sling</groupId>
+            <artifactId>org.apache.sling.jcr.api</artifactId>
```
why is this dep needed? you arent using it

royteeuwen commented on samples/htlblog/pom.xml:
```diff
@@ -0,0 +1,201 @@
+        </dependency>
+        <dependency>
+            <groupId>org.apache.felix</groupId>
+            <artifactId>org.apache.felix.scr.annotations</artifactId>
```
replace these with the osgi annotations

royteeuwen reviewed (commented):
I gave some comments about the pom.xml, the rest looks good 👍

auniverseaway committed:
Updating POM based on PR feedback.

auniverseaway committed:
Merge remote-tracking branch 'upstream/trunk' into SLING-6740_HTLBlog_Sample

auniverseaway committed:
POM cleanup based on PR comments. Add RAT config. Add additional license headers. Small code cleanup.

auniverseaway committed:
Merge branch 'SLING-6740_HTLBlog_Sample' of github.com:auniverseaway/sling into SLING-6740_HTLBlog_Sample

auniverseaway committed:
Minor formatting of license

auniverseaway committed:
Restructure clientlibs to make RAT excludes more concise

auniverseaway commented:
@royteeuwen Addressed all your comments and added in the needed RAT excludes. The Sling Models API / IMPL on Sling 8 is 1.2.2, so I used that instead of 1.3.

royteeuwen commented:
Cool, looks nice :)!

auniverseaway committed:
A bit better hygiene for the models.

auniverseaway committed:
There's no need to close the resourceResolver in this case.
